test/extended/setup_compat.sh

```diff
@@ -0,0 +1,354 @@
+    exit 0
+  fi
+
+  # ensure_iptables_or_die
+
+  function cleanup()
+  {
+    out=$?
+    cleanup_openshift
+    echo "[INFO] Exiting"
+    return $out
+  }
+
+  trap "exit" INT TERM
+  trap "cleanup" EXIT
+  echo "[INFO] Starting server"
+
+  os::util::environment::setup_all_server_vars "test-extended/compatibility"
+  os::util::environment::use_sudo
+  os::util::environment::setup_images_vars
+  reset_tmp_dir
+
+  # If the current system has the XFS volume dir mount point we configure
+  # in the test images, assume to use it which will allow the local storage
+  # quota tests to pass.
+  if [ -d "/mnt/openshift-xfs-vol-dir" ]; then
+    export VOLUME_DIR="/mnt/openshift-xfs-vol-dir"
+  else
+    echo "[WARN] /mnt/openshift-xfs-vol-dir does not exist, local storage quota tests may fail."
+  fi
+
+  os::log::start_system_logger
+
+  # when selinux is enforcing, the volume dir selinux label needs to be
+  # svirt_sandbox_file_t
+  #
+  # TODO: fix the selinux policy to either allow openshift_var_lib_dir_t
+  # or to default the volume dir to svirt_sandbox_file_t.
+  if selinuxenabled; then
+        sudo chcon -t svirt_sandbox_file_t ${VOLUME_DIR}
+  fi
+
+  if [[ -n "${API_SERVER_VERSION:-}" ]] || [[ -n "${CONTROLLER_VERSION:-}" ]]; then
+    configure_versioned_os_server
+  else
+    configure_os_server
+  fi
+  #turn on audit logging for extended tests ... mimic what is done in util.sh configure_os_server, but don't
+  # put change there - only want this for extended tests
+  echo "[INFO] Turn on audit logging"
+  cp ${SERVER_CONFIG_DIR}/master/master-config.yaml ${SERVER_CONFIG_DIR}/master/master-config.orig2.yaml
+  openshift ex config patch ${SERVER_CONFIG_DIR}/master/master-config.orig2.yaml --patch="{\"auditConfig\": {\"enabled\": true}}"  > ${SERVER_CONFIG_DIR}/master/master-config.yaml
+
+  # Similar to above check, if the XFS volume dir mount point exists enable
+  # local storage quota in node-config.yaml so these tests can pass:
+  if [ -d "/mnt/openshift-xfs-vol-dir" ]; then
+      # The ec2 images usually have ~5Gi of space defined for the xfs vol for the registry; want to give /registry a good chunk of that
+      # to store the images created when the extended tests run
+    sed -i 's/perFSGroup: null/perFSGroup: 4480Mi/' $NODE_CONFIG_DIR/node-config.yaml
+  fi
+  echo "[INFO] Using VOLUME_DIR=${VOLUME_DIR}"
+
+  start_etcd
+  start_api_server
+  start_controllers
+
+  if [[ -z "${SKIP_NODE:-}" ]]; then
+    start_node
+  fi
+
+  export KUBECONFIG="${ADMIN_KUBECONFIG}"
+
+  install_registry
+  if [[ -z "${SKIP_NODE:-}" ]]; then
+    wait_for_registry
+    DROP_SYN_DURING_RESTART=1 CREATE_ROUTER_CERT=1 install_router
+    echo "[INFO] Creating image streams"
+    oc create -n openshift -f examples/image-streams/image-streams-centos7.json --config="${ADMIN_KUBECONFIG}"
+  fi
+  return
+}
+
+# start_etcd starts etcd for OpenShift
+function start_etcd() {
```

Comment:
aside from the fact that `hack/util.sh` is too bloated and will be pruned... this seems like it belongs there